### ISSUE TITLE
Simrad TP-32

### ISSUE THREAD

**Issue by MichaKersloot** [SignalK/signalk-autopilot#20]
I'm using a B&G Vulcan, Simrad TP-32 and SignalK to the same N2K network. I'm able to remotely adjust the TP32 from the B&G Vulcan via the N2K network, so there is N2K support on the TP32.

Would you be interested in adding Simrad TP-22/32 support to your autopilot module? And if so, how can I help?

**Comment by tkurki** [comment]
I have exactly the same setup and was planning to do that over the coming winter, if not during this sailing season. I am fairly busy irl right now, but if nothing happens to this feel free to ping me here or in Slack in mid September.

**Comment by MichaKersloot** [comment]
That would be great. For me it's not easy to access the hardware in the winter to generate information or test. But if you have the same setup that wouldn't be a problem ;-) Please feel free to do this at a time that is convenient for you. As happy as I am with opencpn / signalk, it still just a hobby project for a hobby ;-)

**Comment by MichaKersloot** [comment]
Maybe in this project: https://github.com/htool/RaymarineAPtoFakeNavicoAutoPilot there is info you could use.

**Comment by frosenb** [comment]
@tkurki Any progress ;)

**Comment by tkurki** [comment]
No, but a reminder helps bring this forward in the things to do queue!

**Comment by htool** [comment]
We're waiting for a Triton2 autopilot keypad dump to see which packets are used. Then making a webapp that emulates it should hopefully be simple and can serve as an example for others to use it.

**Comment by tkurki** [comment]
I have data from my Vulcan talking to the TP, just need to dig it up.

**Comment by tkurki** [comment]
Hmm. I should probably generate some data with clear purpose and not try to figure out what I was doing with it last summer..

pgns.json:
```
        "PGN":65341,
        "Id":"simnetAutopilotMode",
        "Description":"Simnet: Autopilot Mode",
        "Type":"Single",
        "Complete":false,
        "Missing":[
          "Fields",
          "FieldLengths",
          "Precision"],
```


```
2021-07-13T07:09:24.995Z,6,65341,0,255,8,41,9f,ff,ff,0b,ff,00,00
2021-07-13T07:09:25.026Z,6,65341,0,255,8,41,9f,ff,ff,0c,ff,ff,ff
2021-07-13T07:09:25.051Z,6,65341,0,255,8,41,9f,ff,ff,03,ff,ff,ff
2021-07-13T07:09:25.076Z,6,65341,0,255,8,41,9f,ff,ff,02,ff,e7,82
2021-07-13T07:09:26.028Z,6,65341,0,255,8,41,9f,ff,ff,0d,ff,ff,7f
2021-07-13T07:09:27.083Z,6,65341,0,255,8,41,9f,ff,ff,0d,ff,ff,7f
2021-07-13T07:09:28.137Z,6,65341,0,255,8,41,9f,ff,ff,0d,ff,ff,7f
2021-07-13T07:09:29.192Z,6,65341,0,255,8,41,9f,ff,ff,0d,ff,ff,7f
2021-07-13T07:09:30.251Z,6,65341,0,255,8,41,9f,ff,ff,0d,ff,ff,7f
2021-07-13T07:09:30.277Z,6,65341,0,255,8,41,9f,ff,ff,0b,ff,00,00
2021-07-13T07:09:30.303Z,6,65341,0,255,8,41,9f,ff,ff,0c,ff,ff,ff
2021-07-13T07:09:30.328Z,6,65341,0,255,8,41,9f,ff,ff,03,ff,ff,ff
2021-07-13T07:09:30.353Z,6,65341,0,255,8,41,9f,ff,ff,02,ff,e7,82
2021-07-13T07:09:31.306Z,6,65341,0,255,8,41,9f,ff,ff,0d,ff,ff,7f
2021-07-13T07:09:32.360Z,6,65341,0,255,8,41,9f,ff,ff,0d,ff,ff,7f
2021-07-13T07:09:33.429Z,6,65341,0,255,8,41,9f,ff,ff,0d,ff,ff,7f
2021-07-13T07:09:34.469Z,6,65341,0,255,8,41,9f,ff,ff,0d,ff,ff,7f
2021-07-13T07:09:35.533Z,6,65341,0,255,8,41,9f,ff,ff,0d,ff,ff,7f
2021-07-13T07:09:35.561Z,6,65341,0,255,8,41,9f,ff,ff,0b,ff,00,00
2021-07-13T07:09:35.586Z,6,65341,0,255,8,41,9f,ff,ff,0c,ff,ff,ff
2021-07-13T07:09:35.603Z,6,65341,0,255,8,41,9f,ff,ff,03,ff,ff,ff
2021-07-13T07:09:35.638Z,6,65341,0,255,8,41,9f,ff,ff,02,ff,e7,82
2021-07-13T07:09:36.585Z,6,65341,0,255,8,41,9f,ff,ff,0d,ff,ff,7f
```

**Comment by tkurki** [comment]
I finally finally started with this, based on a recording I made. 

130851 decoding in canboat does not work https://github.com/canboat/canboat/issues/272

**Comment by tkurki** [comment]
https://github.com/SignalK/n2k-signalk/pull/242

**Comment by Leijoma** [comment]
Any updates on this topic? I have tried to emulate a B&G autopilot based on information in the RaymarineAPtoFakeNavicoAutoPilot, but it sends so many messages at strange intervals. I have not been able to replicate it in a way that the Vulcan displays the AP-panel.  I understand that this is not the same as replicating a AP-keypad, but my winter project is to build a simple Autopilot that can act as a B&G or Simrad AP on the network.

**Comment by shlandrews** [comment]
Bumping this one also, keen to utilise this for a Simrad TP22/TP32

**Comment by shlandrews** [comment]
Thanks! Although I upgraded the app, restarted sk and now no plugin config. Reinstalled and still no config. SK v2.13.5

**Comment by GITnonameismyname** [comment]
Thanks for your great efforts! 
Maybe anybody may explain how to set the configuration for a Simrad TP-32. What to put in the plugin config fields? 
I can see my TP-32 in the databrowser under n2k, but it doesn't respond to the remote control commands.
I set only "Autopilot Type" to Simrad NMEA2000, and set a random Name in  "Raymarine NMEA2000 ID".
Anything more to do? 
Thx for any suggestions.
